src/playdate/scoreboards.nim

```diff
@@ -0,0 +1,157 @@
+{.push raises: [].}
+
+import std/importutils
+
+import types {.all.}
+import bindings/[api, types]
+import bindings/scoreboards
+
+# Only export public symbols, then import all
+export scoreboards
+{.hint[DuplicateModuleImport]: off.}
+import bindings/scoreboards {.all.}
+
+type PDScore* = object of RootObj
+  value*: uint32
+  rank*: uint32
+  player*: string
+
+type PDScoresList* = object of RootObj
+  boardID*: string
+  lastUpdated*: uint32
+  scores*: seq[PDScore]
+  # these properties are not implemented yet in the Playdate API
+  # playerIncluded*: uint32
+  # limit*: uint32
+
+type PDBoard* = object of RootObj
+  boardID*: string
+  name*: string
+
+type PDBoardsList* = object of RootObj
+  lastUpdated*: uint32
+  boards*: seq[PDBoard]
+
+type
+  PersonalBestCallback* = proc(score: PDScore, errorMessage: string)
+  AddScoreCallback* = proc(score: PDScore, errorMessage: string)
+  BoardsListCallback* = proc(boards: PDBoardsList, errorMessage: string)
+  ScoresCallback* = proc(scores: PDScoresList, errorMessage: string)
+
+var privatePersonalBestCallbacks = newSeq[PersonalBestCallback]()
+var privateAddScoreCallbacks = newSeq[AddScoreCallback]()
+var privateScoresCallbacks = newSeq[ScoresCallback]()
+var privateBoardsListCallbacks = newSeq[BoardsListCallback]()
```

Comment:
Can you add some documentation around what these are and why they are globals like this?